### PR TITLE
Escape JSON path for definitions

### LINKED ISSUE #1644: $ref autocompletion for definitions whose names contain / or ~
| Q                                   | A
| ----------------------------------- | -------
| Bug or feature request?             | Bug
| Which Swagger/OpenAPI version?      | 2.0
| Which Swagger-Editor version?       | 3.2.5
| How did you install Swagger-Editor? | http://editor.swagger.io
| Which browser & version?            | Chrome
| Which operating system?             | Windows 7

If the name of a definition (or another reusable component) contains `/` or `~`, then the `$ref`erences need to have these characters escaped as `~1` and `~0`, as per https://tools.ietf.org/html/rfc6901#section-3

```yaml
definitions:
  foo/bar: ...
  ab~cd: ...

$ref: '#/definitions/foo~1bar'
$ref: '#/definitions/ab~0cd'
```
However, when the Editor autocompletes such references, it does not encode `/` and `~` in the definition names.

### Demonstration API definition
```yaml
swagger: '2.0'
info:
  version: 0.0.0
  title: test
paths:
  /foo:
    get:
      responses:
        200:
          description: OK
          schema:
            # TODO:
            # replace {} with $ref:
            # then type a space followed by x
            # then press Enter to autocomplete the suggestion
            
            {}
definitions:
  x/Foo~Bar:
    type: object
```

### Expected Behavior
Reference is inserted as `$ref: '#/definitions/x~1Foo~0Bar'`.

### Current Behavior
Reference is inserted as `$ref: '#/definitions/x/Foo~Bar'` which is not a valid reference and causes a resolver error.

reference

### Possible Solution
N/a

### Context
Encountered by a user in https://github.com/swagger-api/swagger-ui/issues/4152.

This is probably not applicable to OpenAPI 3.0 since it does not allow `/` and `~` in the names of [reusable components](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#components-object).

### LINKED ISSUE #1645: Bogus "unused definition" warnings if schema names contain / or ~
| Q                                   | A
| ----------------------------------- | -------
| Bug or feature request?             | Bug
| Which Swagger/OpenAPI version?      | 2.0
| Which Swagger-Editor version?       | 3.2.5
| How did you install Swagger-Editor? | http://editor.swagger.io
| Which browser & version?            | Chrome
| Which operating system?             | Windows 7

Schema names containing `/` or `~` trigger "unused definition" warnings even when these schemas are actually used.

### Demonstration API definition
```yaml
swagger: '2.0'
info:
  version: 0.0.0
  title: test
paths:
  /:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: '#/definitions/x~1Foo'  # '/' is escaped as ~1 as per JSON Pointer rules
         400:
          description: error
          schema:
            $ref: '#/definitions/x~0Bar'  # '~' is escaped as ~0 as per JSON Pointer rules
definitions:
  x/Foo:
    type: object
  x~Bar:
    type: object
```

### Expected Behavior
No "unused definition" warnings.

### Current Behavior
Editor displays warnings that the `x/Foo` and `x~Bar` definitions are not used (which is not true):

warnings

### Possible Solution
N/a

### Context

Editor v2 does not have this issue.

This does not apply to OpenAPI 3.0 because it does not allow `/` and `~` in the [component names](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#componentsObject).

### PR DESCRIPTION
**Pull request by jemerald**
### Description

Escape JSON path for definitions when resolving auto-suggest and when performing validation.

Extracted the `escapeJsonPointerToken` and `unescapeJsonPointerToken` functions to separate file.

Also added `cross-env` dev dependency to allow running the test under windows.

### Motivation and Context

Fixes #1644 and Fixes #1645

### How Has This Been Tested?

Manually tested auto-completion in Chrome.

Added unit test for validation.

### Screenshots (if appropriate):
image

image

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/plugins/editor-autosuggest/spec-selectors.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { createSelector } from "reselect"
 import { Set, Map } from "immutable"
+import { escapeJsonPointerToken } from "../refs-util"
```
Comment (jemerald) [inline src/plugins/editor-autosuggest/spec-selectors.js]: @shockey Is this the preferred way for shared code? Or should it go through the magic plugin system?

Comment (shockey) [reply src/plugins/editor-autosuggest/spec-selectors.js]: This seems fine to me - the plugin system is great for exposing code that someone else may want to override/customize, but I can't imagine someone wanting to write a custom JSON Pointer escaper 😄

**Comment by shockey** [comment]
thanks, @jemerald!
